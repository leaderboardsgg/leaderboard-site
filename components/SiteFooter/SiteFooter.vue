<template>
  <div class="flex flex-wrap rounded bg-gray-100 p-5 text-gray-600">
    <p class="w-full text-center">Copyright 2021</p>
  </div>
</template>
