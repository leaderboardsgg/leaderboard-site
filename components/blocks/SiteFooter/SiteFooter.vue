<script setup lang="ts">
import { computed } from 'vue'

const currentYear = computed(() => new Date().getFullYear())
</script>

<template>
  <div class="site-footer">
    <p class="site-footer__copyright">Copyright {{ currentYear }}</p>
  </div>
</template>

<style lang="postcss" scoped>
.site-footer {
  @apply grid md:grid-cols-3 grid-cols-1 md:gap-y-0 gap-y-5 items-center;
  @apply rounded bg-gray-100 p-5 text-gray-600;

  & .site-footer__copyright {
    @apply md:col-start-2 w-full text-center;
  }
}
</style>
