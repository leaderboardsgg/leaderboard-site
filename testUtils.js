import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
