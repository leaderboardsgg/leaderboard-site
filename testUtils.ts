import Vue, { ComponentOptions } from 'vue';
import { config, RouterLinkStub, VueClass } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render, RenderOptions } from '@testing-library/vue';
import { MockI18nOptions } from '@/types';

export const NuxtLinkStub = Vue.component<
  { tag: string; to: string },
  unknown,
  unknown
>('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data?.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const mockI18n = (options?: MockI18nOptions) => {
  const mockEnglishLocale = { code: 'en', name: 'English' };

  return {
    $i18n: {
      locales: [
        mockEnglishLocale,
        { code: 'es', name: 'Spanish' },
        { code: 'fr', name: 'French' },
        ...(options?.locales ?? []),
      ],
      locale: options?.defaultLocale ?? mockEnglishLocale,
      setLocale: options?.setLocale ?? jest.fn(),
    },
  };
};

export const stubbedRender = (
  Component: VueClass<Vue> | ComponentOptions<Vue>,
  options?: RenderOptions<Vue>,
) => {
  return render(Component, {
    mocks: {
      ...mockI18n(),
      ...options?.mocks,
    },
    stubs: {
      NuxtLink: NuxtLinkStub,
      'svg-icon': true,
      ...options?.stubs,
    },
    ...options,
  });
};

config.mocks.$t = (key: string) => key;
config.mocks.localeRoute = (route: string) => route;

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
