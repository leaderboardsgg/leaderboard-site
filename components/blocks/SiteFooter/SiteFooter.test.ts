import SiteFooter from './SiteFooter.vue'
import { stubbedRender } from '@/testUtils'

describe('<SiteFooter />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(SiteFooter)

    unmount()
  })

  it('renders correctly with the copyright info', () => {
    const { getByText } = stubbedRender(SiteFooter)
    const currentYear = new Date().getFullYear()

    expect(getByText(`Copyright ${currentYear}`)).toBeInTheDocument()
  })
})
