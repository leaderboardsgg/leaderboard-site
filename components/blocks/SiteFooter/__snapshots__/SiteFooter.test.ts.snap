// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteFooter /> renders correctly 1`] = `
<div
  class="site-footer"
>
  <p
    class="site-footer__copyright"
  >
    Copyright 2021
  </p>
   
  <select
    class="language-selector site-footer__language-selector"
    data-testid="language-selector"
    role="listbox"
  >
    <option
      value="en"
    >
      
    English
  
    </option>
    <option
      value="es"
    >
      
    Spanish
  
    </option>
    <option
      value="fr"
    >
      
    French
  
    </option>
  </select>
</div>
`;
